first_order_model/shrink_util.py

```diff
@@ -1060,43 +1109,49 @@ def compute_deletion(layer_graph,
     for mirror in layer_graph[layer].mirror:
         # Start a new custom deletion for the mirror. The only other reason is tie, which is treated a little differently.
         if reason != 'mirror':
-            custom_deletions.append((mirror, copy.copy(deletions[layer]), 'mirror'))
+            filters_to_recursively_remove.append((mirror, copy.copy(deletions[layer]), 'mirror'))
             print("Starting a mirrorred deletion")
 
     # Figure out what you need to delete from the inputs of the following convs, and the relevant portions of batchnorms.
-    # Following layers contains any layeres who can be affected by the deletion in layer_graph[layer]
+    # Following layers contains any layers who can be affected by the deletion in layer_graph[layer]
     for following_layer in following_layers:
         # If we have already addressed this layer, continue
         if following_layer in deletions:
             continue
 
         deletions[following_layer] = []
         
-        # Counts up for each index in the inputs for this layer
-        counter = 0
+        # base counts up for each index in the inputs for this layer
+        # This is to solve the following problem:
+        # If layer a, and b get concatted into the inputs of c, then we want to note that
+        # deleting from b should only affect its corresponding inputs in c.
+        # We use this index to track the "base" of an input. So if layer a has length 20, layer b has length 10
```

Comment:
Same issue as before that I don't understand how if b has only length 10, what does affecting indices beyond 20 mean? Can you improve this comment?

Reply:
I  cleaned this up